src/test_workflow/smoke_test/smoke_tests_spec/opensearch.yml

```diff
@@ -0,0 +1,30 @@
+# yamllint disable
```

Comment:
How about storing these files under manifests folder per version?

```
manifests/2.18.0/smoke_test/opensearch.yml
```

Reply:
At this point, i don't feel it's necessary to keep these files distinct by version. So far the API requests in this file are generic and not version specific. In addition, any of the valid API path can be filtered in the API-spec file for the first version it was introduced.

Reply:
For new APIs that are being added how do you propose to manage that?

Reply:
The version introduced will be specified in the API-spec file and we can check the version and compare it with the current version.

Reply:
Sorry for not being clear, what I meant was if we introduce new APIs in say 2.19.0 would we be using same spec file that we used in 2.18.0?
Wondering how api-spec repo is managing that today.

Reply:
We have `x-version-add` in api-spec nowadays. For example, https://github.com/opensearch-project/opensearch-api-specification/blob/ccce0656eecca26f18d4ba06ece9155d96b15d7a/spec/namespaces/search_pipeline.yaml#L11

Reply:
I haven't added the check in this workflow yet but I will consider it as an enhancement that should be added.

Reply:
So for new x-version-add (2.9, 2.10, 2.11), how would the spec look like?

Reply:
This component spec will still be the same. We check on the test workflow side. 
The workflow extracts the api path from this component spec, checks current cluster version vs the x-version-add from api-spec and run checks if current version is later than added version.

Reply:
if not differentiated on version it will become a problem when 3.0 is being prepared to release and has breaking changes in existing apis. 
You don't have to separate at each minor but atleast at 2.x and 3.x. Minor versions are backward compatible so it is safe to assume there will be no breaking changes in the apis. 
We don't have to worry about introducing new apis in minor versions as smoke tests are expected to run for versions to be released and not already released versions.

Reply:
I'm okay if openapi handles the version check and returns appropriate response. 
Say, version 3.0 made a change to `_cat/indices` api request/response and you run this workflow for both 2.x and 3.x for a new release, how will that be handled?

Reply:
Those changes should also be handled in the openapi spec files.

Reply:
@gaiksaya @rishabh6788 I updated these component spec by versions to accommodate any major version breaking API changes. Please help review it again. Thanks!